src/passes/DeadArgumentElimination.cpp

```diff
@@ -176,50 +189,82 @@ struct DAE : public Pass {
   bool optimize = false;
 
   void run(Module* module) override {
+    DAEFunctionInfoMap infoMap;
+    // Ensure all entries exist so the parallel threads don't modify the data
+    // structure.
+    for (auto& func : module->functions) {
+      infoMap[func->name];
+    }
+    // The null name represents module-level code (not in a function).
+    infoMap[Name()];
+
     // Iterate to convergence.
     while (1) {
-      if (!iteration(module)) {
+      if (!iteration(module, infoMap)) {
         break;
       }
     }
   }
 
-  bool iteration(Module* module) {
+  bool iteration(Module* module, DAEFunctionInfoMap& infoMap) {
     allDroppedCalls.clear();
 
-    DAEFunctionInfoMap infoMap;
-    // Ensure they all exist so the parallel threads don't modify the data
-    // structure.
-    for (auto& func : module->functions) {
-      infoMap[func->name];
+#if DAE_DEBUG
+    // Enable this path to mark all contents as stale at the start of each
+    // iteration, which can be used to check for staleness bugs. Note, though,
```

Comment:
Can you clarify what kind of staleness bugs you're referring to here? Is it when things that should have been marked stale were not? And then you'd detect that because enabling this code path would lead to different results?

Reply:
Yes, exactly.

I added a comment now.